jira_integration_example/utilities/alert.py

```diff
@@ -0,0 +1,250 @@
+    used as monitoring data for testing purposes.
+
+    Attributes:
+        _project_id: The id of the project to store metrics
+        _client: A monitoring_v3.MetricServiceClient instance to modify metric data
+    """
+
+    def __init__(self, project_id):
+        self._project_id = project_id
+        self._client = monitoring_v3.MetricServiceClient()
+        self._project_name = self._client.project_path(self._project_id)
+
+
+    def create_custom_metric(self, metric_name):
+        """Creates a custom metric with the given metric name.
+
+        Args:
+            metric_name: The name of the metric
+        """
+        descriptor = monitoring_v3.types.MetricDescriptor()
+        descriptor.type = 'custom.googleapis.com/' + metric_name
+        descriptor.metric_kind = (
+            monitoring_v3.enums.MetricDescriptor.MetricKind.GAUGE)
+        descriptor.value_type = (
+            monitoring_v3.enums.MetricDescriptor.ValueType.DOUBLE)
+        descriptor.description = 'A custom metric meant for testing purposes'
+        descriptor = self._client.create_metric_descriptor(self._project_name, descriptor)
+        print(f'Created {descriptor.name}.')
+
+
+    def get_custom_metric(self, metric_name):
+        """Get the custom metric with the given metric name.
+
+        Args:
+            metric_name: the name of the metric descriptor
+
+        Raises:
+            CustomMetricNotFoundError: if the metric doesn't exist
+        """
+        try:
+            return self._client.get_metric_descriptor(f'projects/{self._project_id}/metricDescriptors/custom.googleapis.com/{metric_name}')
+        except NotFound:
+            raise CustomMetricNotFoundError(f'{metric_name} does not exist.')
+
+
+    def append_to_time_series(self, metric_name, point_value):
+        """Add a data point with the given point value to the specified metric.
+
+        Args:
+            metric_name: The name of the metric to modify
+            point_value: The value of the data point to add
+        """
+        series = monitoring_v3.types.TimeSeries()
+        series.metric.type = 'custom.googleapis.com/' + metric_name
+        series.resource.type = 'gce_instance'
+        series.resource.labels['instance_id'] = '1234567890123456789'
+        series.resource.labels['zone'] = 'us-central1-f'
+        point = series.points.add()
+        point.value.double_value = point_value
+        now = time.time()
+        point.interval.end_time.seconds = int(now)
+        point.interval.end_time.nanos = int(
+            (now - point.interval.end_time.seconds) * 10**9)
+        self._client.create_time_series(self._project_name, [series])
+
+
+    def delete_custom_metric(self, metric_name):
+        """Delete the given custom metric.
+
+        Args:
+            metric_name: Name of metric to delete
+        """
+        descriptor_name = f'{self._project_name}/metricDescriptors/custom.googleapis.com/{metric_name}'
```

Comment:
As pointed out in my example, this could be simplified by reusing the original metric descriptor. For example:

```
descriptor = monitoring_client.create_metric_descriptor(...)
...
...
monitoring.client.delete_metric_descriptor(descriptor.name)
```